snapfaas/src/sched/rpc.rs

```diff
@@ -0,0 +1,114 @@
+use std::net::{TcpStream, SocketAddr};
+use std::thread;
+use std::collections::HashMap;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use serde::{Serialize, Deserialize};
+
+use super::Error;
+use super::message;
+use super::message::{Request, Response};
+use super::message::request::Kind as ReqKind;
+
+/// RPC calls
+#[derive(Debug, Clone)]
+pub struct Scheduler {
+    sa: SocketAddr,
+}
+
+impl Scheduler {
+    pub fn new(sa: SocketAddr) -> Self {
+        Scheduler { sa }
+    }
+
+    fn connect(&self) -> Result<TcpStream, Error> {
+        TcpStream::connect(&self.sa).map_err(|e| Error::StreamConnect(e))
+    }
+
+    /// This method is for workers to retrieve a HTTP request, and
+    /// it is supposed to block if there's no further HTTP requests
+    pub fn get(&self) -> Result<Response, Error> {
+        let mut stream = self.connect()?;
```

Comment:
So this will create a _new_ TcpStream each time there is a request. Instead, wouldn't it be better to either have the scheduler store a persistent TcpStream or a pool of (reconnecting) TcpStreams?

Reply:
Right, it would be better to have a persistent connection. I'll change it accordingly as well.